File: 1863/solution.ts
function subsetXORSum(nums: number[]): number {
  let res = 0;
  let subset = 0;
  const backtrack = (i: number) => {
    if (i == nums.length) {
      res += subset;
      return;
    }
    subset ^= nums[i];
    backtrack(i + 1);
    subset ^= nums[i];
    backtrack(i + 1);
  };

  backtrack(0);
  return res;
}

File: 1863/solution.java
class Solution {
    int res = 0;
    int subset = 0;

    public int subsetXORSum(int[] nums) {
        backtrack(0, nums);
        return res;
    }

    private void backtrack(int i, int[] nums) {
        if (i == nums.length) {
            res += subset;
            return;
        }
        subset ^= nums[i];
        backtrack(i + 1, nums);
        subset ^= nums[i];
        backtrack(i + 1, nums);
    }
}

// time O(2^n)
// space O(n)